src/faction_camp.cpp

```diff
@@ -271,11 +269,24 @@ static basecamp *get_basecamp( npc &p )
     if( bcp ) {
         return bcp;
     }
-    g->m.add_camp( p.pos(), "faction_camp" );
+    std::string camp_name;
```

Comment:
Please do all of this inside define_camp(), and don't do it if g->m.camp_at() fails because there is a corner case where someone gets the camp manager dialogue while not actually being at a faction camp and that should fail, not spontaneously create a faction camp.

Reply:
Rethinking this again:
call `g->m.add_camp()` with a generic name like the code did, and then rename inside of `define_camp()`.